dyndbmutex/dyndbmutex.py

```diff
@@ -145,8 +152,10 @@ def prune_expired(self, lockname, caller):
 
 class DynamoDbMutex:
 
-    def __init__(self, name, holder=str(uuid.uuid4()),
+    def __init__(self, name, holder=None,
                  timeoutms=30 * 1000, region_name='us-west-2'):
+        if holder is None:
```

Comment:
The default values of keyword arguments are evaluated at function definition time, not function call time. I wasn't sure what your original intention was with the holder. The way it was before all instances of this class in the same process (Python interpreter) would share the same holder unless they explicitly specified one to the constructor. I find that counter-intuitive.

```
import uuid
def x(x=str(uuid.uuid4())):
    return x
x()
'f8fd8b3a-28cb-4f6c-8e0a-9eb45f75ae89'
x()
'f8fd8b3a-28cb-4f6c-8e0a-9eb45f75ae89'
def y(y=None):
    return str(uuid.uuid4()) if y is None else y
y()
'ce0a0f00-cfc5-40ea-a2db-6d56b4115e33'
y()
'e7711eaf-851e-4c91-a5e7-a785942ab353'
```

Reply:
didn't consider that it would be used in the same interpreter!